ms2query/spectrum_processing.py

```diff
@@ -0,0 +1,162 @@
+                "n_required_below_1000": 5,
+                "intensity_from": 0.001,
+                "max_mz_required": 1000.0,
+                }
+
+    # Set default parameters or replace by **settings input
+    for key in defaults:
+        if key not in settings:
+            settings[key] = defaults[key]
+    return settings
+
+
+def spectrum_processing_minimal(spectrum: SpectrumType,
+                                **settings: Dict[str, Union[int, float]]) \
+    -> Union[SpectrumType, None]:
+    """Minimal necessary spectrum processing that is required by MS2Query.
+    This mostly includes intensity normalization and setting spectrums to None
+    when they do not meet the minimum requirements.
+
+    Args:
+    ----------
+    spectrum:
+        Spectrum to process
+    mz_from
+        Set lower threshold for m/z peak positions. Default is 10.0.
+    n_required_below_1000
+        Number of minimal required peaks with m/z below 1000.0Da for a spectrum
+        to be considered. Spectrums not meeting this criteria will be set to None.
+    intensity_from
+        Set lower threshold for peak intensity. Default is 0.001.
+    max_mz_required
+        Only peaks <= max_mz_required will be counted to check if spectrum
+        contains sufficient peaks to be considered.
+    """
+    settings = set_minimal_processing_defaults(**settings)
+    spectrum = normalize_intensities(spectrum)
+    spectrum = select_by_intensity(spectrum, intensity_from=settings["intensity_from"])
+    spectrum = select_by_mz(spectrum, mz_from=settings["mz_from"], mz_to=np.inf)
+    spectrum = require_peaks_below_mz(spectrum, n_required=settings["n_required_below_1000"],
+                                      max_mz=settings["max_mz_required"])
+    return spectrum
+
+def set_spec2vec_defaults(**settings: Dict[str, Union[int, bool]]) \
+        -> Dict[str, Union[int, bool]]:
+    """Set spec2vec default argument values"(where no user input is given)".
+
+    Args
+    ----------
+    **settings:
+        Change default settings
+    """
+    defaults = {"mz_from": 0,
+                "mz_to": 1000,
+                "n_required": 1,
+                "ratio_desired": 0.5,
+                "intensity_from": 0.001,
+                "n_max": 1000,
+                "loss_mz_from": 5.0,
+                "loss_mz_to": 200.0,
+                "intensity_weighting_power": 0.5,
+                "allowed_missing_percentage": 10.0,
+                }
+
+    # Set default parameters or replace by **settings input
+    for key in defaults:
+        if key not in settings:
+            settings[key] = defaults[key]
+    return settings
+
+
+def spectrum_processing_s2v(spectrum: SpectrumType,
+                            **settings: Dict[str, Union[int, bool]]) \
+    -> Union[SpectrumType, None]:
+    """Spectrum processing required for computing Spec2Vec scores.
+
+    Args:
+    ----------
+    spectrum:
+        Spectrum to process
+    mz_from
```

Comment:
A lot of these variables are not used anymore, so they can be removed.

Reply:
Thanks. Removed the old ones.